CMakeLists.txt

```diff
@@ -394,16 +385,44 @@ endif(WITH_TESTS)
 if(WITH_COVERAGE)
     # Include code coverage, use with -DCMAKE_BUILD_TYPE=Debug
     include(CodeCoverage)
-    set(COVERAGE_GCOVR_EXCLUDES
+    set(COVERAGE_EXCLUDES
             "\\(.+/\\)?tests/.\\*"
+            "\\(.+/\\)?build/.\\*"
+            "\\(.+/\\)?thirdparty/.\\*"
+            "\\(.+/\\)?CMakeFiles/.\\*"
+            "src/main.cpp"
             ".\\*/moc_\\[^/\\]+\\.cpp"
             ".\\*/ui_\\[^/\\]+\\.h"
-            "\\(.+/\\)?zxcvbn/.\\*")
+            ".\\*/\\[^/\\]+_autogen/.\\*"
+            "\\(.+/\\)?zxcvbn/.\\*"
+            "/Applications/.\\*"
+            "/opt/.\\*")
     append_coverage_compiler_flags()
-    setup_target_for_coverage_gcovr_html(
-            NAME coverage
-            EXECUTABLE $(MAKE) && $(MAKE) test
-    )
+
+    if(WITH_COVERAGE AND CMAKE_COMPILER_IS_CLANGXX)
+        if(APPLE AND WITH_APP_BUNDLE)
+            set(MAIN_BINARIES
+                    "${CMAKE_BINARY_DIR}/src/${BIN_INSTALL_DIR}/${PROGNAME}"
```

Comment:
We really should be using generator expressions for these https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_FILE

Reply:
Good idea.

Reply:
Done. Much cleaner already.